### DISCUSSION TITLE
Purchase code signing certificates?

### DISCUSSION THREAD

**Discussion by dogweather** [jamulussoftware/jamulus#1042] [General]
I do Mac development, and can help with this:

<img width="532" alt="Screen Shot 2020-08-27 at 12 09 10 PM" src="https://user-images.githubusercontent.com/150670/91479308-78d4c000-e85e-11ea-8231-6947fca85690.png">

**Comment by ann0see** [comment]
You need to right click/ctrl click + click on open.
See the wiki for more information. https://jamulus.io/wiki/Installation-for-Macintosh#jamulus-cant-be-opened-because-the-developer-cannot-be-verified
Edit: I didn't read correctly and misinterpreted the post.

**Comment by dogweather** [comment]
True, but I'd like to run the software w/out making a security exception.

<img width="581" alt="Screen Shot 2020-08-27 at 12 14 17 PM" src="https://user-images.githubusercontent.com/150670/91479552-d6690c80-e85e-11ea-8f5b-23a73a19a156.png">

**Comment by ann0see** [comment]
Ah. Sorry. I thought you had a problem opening the app.

**Comment by dogweather** [comment]
:-) No, I'm thinking about helping out by doing the signing and verification with Apple.

**Comment by ann0see** [comment]
That sounds good. Let's wait what corrados says. The same problem occurs on Windows with Smart Screen since probably Jamulus doesn't get installed very often.

**Comment by dogweather** [comment]
@corrados does the Mac dev? I can contribute the $99 Apple dev fee if that's what it takes. I'm also happy to actually do the code signing and other work.

**Comment by corrados** [comment]
Sure, you could create the Mac binary releases if you like. What I do is simply call the script:
`sh mac/deploy_mac.sh`

**Comment by dogweather** [comment]
Great, I'll give it a shot.

**Comment by CedricArbuckle** [comment]
I know you all do this in your spare time.  But a group of us are trying to figure out how to sing together (at a distance)  Some of us have macs and get the first message posted ( not the second one)  I see dogweather is working on this and wonder if this will be a hard problem to solve and will be weeks or months in the solving or will it be more like days?

**Comment by ann0see** [comment]
I wouldn’t say it’s hard to "solve" the problem. In the meantime do the right click + Open trick I mentioned in the first answer.

**Comment by WolfganP** [comment]
> I know you all do this in your spare time. But a group of us are trying to figure out how to sing together (at a distance) Some of us have macs and get the first message posted ( not the second one) I see dogweather is working on this and wonder if this will be a hard problem to solve and will be weeks or months in the solving or will it be more like days?

Are you using the workaround at https://github.com/corrados/jamulus/wiki/Installation-for-Macintosh ?
Whatever resolution comes from @dogweather or @corrados it will streamline the installation process, but the app and functionality will be the same as it's today via the side install workaround.

**Comment by CedricArbuckle** [comment]
thank you.  It worked to get the app open.

**Comment by ann0see** [comment]
Any updates?

**Comment by JWuStudio** [comment]
Thanks for your work, this was effective as of 3.5.11.
But the message is back today after installing 3.5.12. just fyi.

**Comment by corrados** [comment]
@dogweather What is the status of your investigation? Did you succeed with the signing?

**Comment by JWuStudio** [comment]
@corrados I think it is working now, again. I reinstalled it today, and it just said "“JamulusServer” is an app downloaded from the Internet. Are you sure you want to open it?", instead of the "developer cannot be verified" message. do not know if @dogweather did anything. Not an expert in this field either.

**Comment by dogweather** [comment]
@corrados I'm sorry, I haven't looked into it yet. I haven't actually tried out Jamulus either -- it definitely looks promising to me; I just haven't gotten to it.

**Comment by CedricArbuckle** [comment]
We are succeeding in our singing, Thank you.

**Comment by ann0see** [comment]
The same problem occurs on Windows (SmartScreen): with MS edge you can't even download it, unless you trust it. 

![JamulusMSEdgeSmartScreenBlock](https://user-images.githubusercontent.com/20726856/96837202-5322fc00-1446-11eb-9ab0-b1f7383001e3.png)

**Comment by ann0see** [comment]
Any chance it comes into the next release?

**Reply by emlynmac** [reply]
I've got it almost done. Just a couple of things to iron out....

**Reply by emlynmac** [reply]
Wrapped up in PR https://github.com/jamulussoftware/jamulus/pull/1856

**Reply by ann0see** [reply]
Great! Thanks.
